### ISSUE TITLE
set different DPS for power consumption feature

### ISSUE THREAD

**Issue by DSchumacher2104** [mileperhour/localtuya-homeassistant#1]
First, thank you for this great Integration!

I have 4 different tuya-plugs and they use different DPS for power consumption feature. Is it possible to extend this integration in a way that makes it possible to set the DPS for power-monitoring individually per device?

I have the same as yours with RGB-light. The DPS are the same as written in your switch.py. The other plugs have different DPS:

OxaOxe double-outlet  DPS: {"1":true,"2":true,"7":95,"8":48,"9":2267} PM here is 7-9, switch 1+2!
gosund 16A DPS:  {"1":true,"2":0,"4":19,"5":8,"6":2280}  PM here is 4-6, switch 1
vodool 10A DPS: {"1":true,"2":0,"4":40,"5":45,"6":2280} PM here also 4-6, switch 1

Thank you!

**Comment by NameLessJedi** [comment]
I had same issue and came up with some simple solution. Have a look at https://github.com/NameLessJedi/localtuya-homeassistant (I've got this running for last 2 days)
@mileperhour I'll create pull-request later on

**Comment by mileperhour** [comment]
Hey, @DSchumacher2104 , please, use the implementation created by @NameLessJedi, as it will allow you to assign the different DPS of every switch in the single declaration for every them.

The other "classic" approach for this issue will be to duplicate the 'localtuya' integration into another one i.e. 'localtuya2': where you will do the modifications (DPS assignments) to this another switch (or series of switches). But please, I do insist to use the other implementation instead for this kind of use, as it's cleaner, and the extra code is not a disadvantage, but a more versatile integration in this case.

@NameLessJedi, thanks for your update and tweaks, I have not all the free time I would like, lately, to fix this things and mantain the integration as updated as it must.

I would be glad if any  other user would try to complete the RGB implementation I marked on the 'to-do' list. Really I think it would be easy to send the appropiate DPS "scene" to charge it before turning on the Led (it really don't differ a lot from sending a "true" on the appropiate register-DPS as for turning on). But other time, I have not enough free time to do this approach. The full "RGB" control may be imported or even charged as a "light" device in the "second" switch, using other code integration for other people -i will swear i saw some examples but I really can't put you on the right direction right now-.

In other words: Please, all of you feel free to tweak, update, clone or add more functionality to the code.

Cheers.

**Comment by NameLessJedi** [comment]
@mileperhour glad you liked it :) Unfortunately can't really help with RGB. I've got only tuya plugs. (all bulbs I have are yeelights and at this moment I don't need more).
Pretty soon I should get tuya based wall switches and will gladly add them (if needed).

**Comment by DSchumacher2104** [comment]
Okay, thank you for this imformation. I have a question about the configuration of this add-on. According to you instructions it has to be:
```
       switch:
         - platform: localtuya
           host: 192.168.1.251
           local_key: 1234567890abcdef
           device_id: abcdef1234567890abcdef
           switches:
             switch1:
             friendly_name: TUYA_LED
             id: 1
           switch2:
             friendly_name: TUYA_SW01
             id: 101
```
I wonder about the different spaces between `switch1:` and `switch2:`. Is this a typo? Another point is that i want to name entity_id's and friendly_name's different, like:
```
entity_id: switch.outlet_cofeemaker
friendly_name: Coffeemaker Power
```
Is this possible? Is `switch1:` the name of the entity_id?

**Comment by NameLessJedi** [comment]
It's typo. It should look like this:
```
switch:
  - platform: localtyua
   host: 192.168.1.1
    local_key: 12345
    device_id: 12345
    switches:
      some_switch1:
        id: 1
        friendly_name:  Some Switch 1
      switch2:
        id: 2
```
This also answers the other question: yes `some_switch1` and `switch1` are `entity_id`
